includes/services/OpenLayers/jquery.openlayers.js

```diff
@@ -64,45 +66,56 @@
-					groups++;
-					curLayer.id = 'markerLayer' + groups;
-					// define default icon, one of ten in different colors, if more than ten layers, colors will repeat:
-					curLayer.defaultIcon = mw.config.get( 'egMapsScriptPath' ) + '/includes/services/OpenLayers/OpenLayers/img/marker' + ( ( groups + 10 ) % 10 ) + '.png';
-					map.addLayer(curLayer);
-					groupLayers[ location.group ] = curLayer;
-				} else {
-					// if markers of this group exist already, they have an own layer already
-					var curLayer = groupLayers[ location.group ];
-				}
+			if (bounds != null) map.zoomToExtent(bounds); // If a bounds object has been created, use it to set the zoom and center.
+		};
 
-				location.lonlat = new OpenLayers.LonLat(location.lon, location.lat);
+		this.addMarker = function (markerData) {
+			markerData.group = !markerData.hasOwnProperty('group') ? '' : markerData.group;
+			// Create a own marker-layer for the marker group:
+			if (!this.groupLayers[ markerData.group ]) {
+				// in case no group is specified, use default marker layer:
+				var layerName = markerData.group != '' ? markerData.group : mw.msg('maps-markers');
+				var curLayer = new OpenLayers.Layer.Markers(layerName);
+				this.groups++;
+				curLayer.id = 'markerLayer' + this.groups;
+				// define default icon, one of ten in different colors, if more than ten layers, colors will repeat:
+				curLayer.defaultIcon = mw.config.get( 'egMapsScriptPath' ) + '/includes/services/OpenLayers/OpenLayers/img/marker' + ( ( this.groups + 10 ) % 10 ) + '.png';
+				map.addLayer(curLayer);
+				this.groupLayers[ markerData.group ] = curLayer;
+			} else {
+				// if markers of this group exist already, they have an own layer already
+				var curLayer = this.groupLayers[ markerData.group ];
+			}
 
-				if (!hasImageLayer) {
-					location.lonlat.transform(new OpenLayers.Projection("EPSG:4326"), new OpenLayers.Projection("EPSG:900913"));
-				}
+			markerData.lonlat = new OpenLayers.LonLat(markerData.lon, markerData.lat);
 
-				if (bounds != null) bounds.extend(location.lonlat); // Extend the bounds when no center is set.
-				var marker = this.getOLMarker(curLayer, location);
-				this.markers.push({
-					target:marker,
-					data:location
-				});
-				curLayer.addMarker(marker); // Create and add the marker.
+			if (!hasImageLayer) {
+				markerData.lonlat.transform(new OpenLayers.Projection("EPSG:4326"), new OpenLayers.Projection("EPSG:900913"));
 			}
 
-			if (bounds != null) map.zoomToExtent(bounds); // If a bounds object has been created, use it to set the zoom and center.
+			var marker = this.getOLMarker(curLayer, markerData);
+			this.markers.push({
+				target:marker,
+				data:markerData
+			});
+			curLayer.addMarker(marker); // Create and add the marker.
+		};
+
+		this.removeMarkers = function () {
+			var map = this.map;
+			$.each(this.groupLayers, function(index, layer) {
+				map.removeLayer(layer);
+			});
```

Comment:
First I wanted to remove markers directly but, some layers also need to be removed. The `removeMarkers` function only removes `groupLayers` they are only used for markers. Lines, etc. have their own layers like `lineLayer`.